### PR TITLE
[Docs] Catalog promotion label customization

### PR DESCRIPTION
**Pull request by AdamKasp**
| Q               | A
| --------------- | -----
| Branch?         | master
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| License         | MIT

should be merged after https://github.com/Sylius/Sylius/pull/13389/files

Screenshot 2021-12-14 at 08 49 26
Screenshot 2021-12-14 at 08 49 34
Screenshot 2021-12-14 at 08 49 37

### CODE REVIEW

**Review by Zales0123 — COMMENTED** [review]
If we need to override so much code just to customize the label... it seems that something is wrong :( Maybe we can extract some template and make it simpler and more generic?

File: docs/cookbook/promotions/customize-catalog-promotion-labels.rst [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+==========================================
+
+To customize catalog promotion labels only what need to do is a overwrite proper template
+
+.. tip::
+
+    Learn more about template customization :doc:`here </customization/template>`.
+
+Changing styles of Catalog Promotion labels.
+--------------------------------------------
```
Comment (CoderMaggie) [inline docs/cookbook/promotions/customize-catalog-promotion-labels.rst:11]:
```suggestion
Change the catalog promotion labels' styles
-------------------------------------------
```

File: docs/cookbook/promotions/customize-catalog-promotion-labels.rst [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+
+In default Sylius we show catalog promotion labels in 2 places: on product index and product show page. So we need to customize  2 related templates:
```
Comment (CoderMaggie) [inline docs/cookbook/promotions/customize-catalog-promotion-labels.rst:13]:
```suggestion
By default Sylius displays catalog promotion labels in two locations: on the products' index on product's card and also on the product show page. That's why you'll need to customize two related templates:
```

File: docs/cookbook/promotions/customize-catalog-promotion-labels.rst [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+
+For product index, we will customize template `@SyliusShopBundle/Product/Box/_content.html.twig`
```
Comment (CoderMaggie) [inline docs/cookbook/promotions/customize-catalog-promotion-labels.rst:15]:
```suggestion
For product index, we will customize the ``@SyliusShopBundle/Product/Box/_content.html.twig`` template
```

File: docs/cookbook/promotions/customize-catalog-promotion-labels.rst [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+by creating a file ``templates/bundles/SyliusShopBundle/Product/Box/_content.html.twig`` with content:
```
Comment (CoderMaggie) [inline docs/cookbook/promotions/customize-catalog-promotion-labels.rst:16]:
```suggestion
by overriding it with your own ``templates/bundles/SyliusShopBundle/Product/Box/_content.html.twig`` that will have such content:
```

File: docs/cookbook/promotions/customize-catalog-promotion-labels.rst [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+    </div>
+
+After changes:
+
+.. image:: ../../_images/cookbook/customization-catalog-promotion/catalog_promotion_label_product_index.png
+    :align: center
+
+
+For product show page we will customize template `@SyliusShopBundle/Product/Show/_price.html.twig`
+by creating a file ``templates/bundles/SyliusShopBundle/Product/Show/_price.html.twig`` with content:
```
Comment (CoderMaggie) [inline docs/cookbook/promotions/customize-catalog-promotion-labels.rst:61]:
```suggestion
For product show page you'll need to override the ``@SyliusShopBundle/Product/Show/_price.html.twig`` template with ``templates/bundles/SyliusShopBundle/Product/Show/_price.html.twig`` looking like that:
```

File: docs/cookbook/promotions/customize-catalog-promotion-labels.rst [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+
+    <span class="ui huge header" id="product-price" {{ sylius_test_html_attribute('product-price', money.calculatePrice(variant)) }}>
+        {{ money.calculatePrice(variant) }}
+    </span>
+
+After changes:
+
+.. image:: ../../_images/cookbook/customization-catalog-promotion/catalog_promotion_label_product_show.png
+    :align: center
+
```
Comment (CoderMaggie) [inline docs/cookbook/promotions/customize-catalog-promotion-labels.rst:97]:
```suggestion

Of course, you can modify the styles how you need, also in your stylesheet.
```

File: docs/cookbook/promotions/customize-catalog-promotion-labels.rst [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+Display catalog promotion labels on photo instead of under it
```
Comment (CoderMaggie) [inline docs/cookbook/promotions/customize-catalog-promotion-labels.rst:98]:
```suggestion
Display catalog promotion labels on the product image
```

File: docs/cookbook/promotions/customize-catalog-promotion-labels.rst [hunk 8]
```diff
@@ -0,0 +1,167 @@
...
+-------------------------------------------------------------
+
+For changing place of catalog promotion label on product index, we need to overwrite 2 templates:
```
Comment (CoderMaggie) [inline docs/cookbook/promotions/customize-catalog-promotion-labels.rst:101]:
```suggestion
In order to change the location of catalog promotion label on the product's card, we need to overwrite two templates. We will remove the label from underneath the photo in product card template and then add it on the product's image.
```

File: docs/cookbook/promotions/customize-catalog-promotion-labels.rst [hunk 9]
```diff
@@ -0,0 +1,167 @@
...
+
+template `@SyliusShopBundle/Product/Box/_content.html.twig`
+by creating a file ``templates/bundles/SyliusShopBundle/Product/Box/_content.html.twig`` with content:
```
Comment (CoderMaggie) [inline docs/cookbook/promotions/customize-catalog-promotion-labels.rst:104]:
```suggestion
Override the ``@SyliusShopBundle/Product/Box/_content.html.twig`` template with
``templates/bundles/SyliusShopBundle/Product/Box/_content.html.twig``:
```

File: docs/cookbook/promotions/customize-catalog-promotion-labels.rst [hunk 10]
```diff
@@ -0,0 +1,167 @@
...
+                {% set originalPrice = money.calculateOriginalPrice(product|sylius_resolve_variant) %}
+                {% if price != originalPrice %}
+                    <div class="sylius-product-original-price" {{ sylius_test_html_attribute('product-original-price') }}><del>{{ originalPrice }}</del></div>
+                {% endif %}
+                <div class="sylius-product-price" {{ sylius_test_html_attribute('product-price') }}>{{ price }}</div>
+            {% endif %}
+        </div>
+    </div>
+
+And template `@SyliusShopBundle/Product/_mainImage.html.twig` by creating a file ``templates/bundles/SyliusShopBundle/Product/_mainImage.html.twig`` with content:
```
Comment (CoderMaggie) [inline docs/cookbook/promotions/customize-catalog-promotion-labels.rst:135]:
```suggestion
And the ``@SyliusShopBundle/Product/_mainImage.html.twig`` with ``templates/bundles/SyliusShopBundle/Product/_mainImage.html.twig``:
```

File: docs/cookbook/promotions/customize-catalog-promotion-labels.rst [hunk 11]
```diff
@@ -0,0 +1,167 @@
...
+                    </div>
+                </div>
+            {% endfor %}
+        </div>
+    </div>
+
+After changes:
+
+.. image:: ../../_images/cookbook/customization-catalog-promotion/catalog_promotion_repositioned_label.png
+    :align: center
```
Comment (CoderMaggie) [inline docs/cookbook/promotions/customize-catalog-promotion-labels.rst:167]:
```suggestion
    :align: center
    
    Well done! Now you can do anything you want with the catalog promotion labels.
```

🎉

**Review by CoderMaggie — CHANGES_REQUESTED** [review]
Let's maybe rename the file to be simply:
`docs/cookbook/promotions/catalog-promotion-labels.rst`

File: docs/cookbook/promotions/customize-catalog-promotion-labels.rst [hunk 12]
```diff
@@ -0,0 +1,113 @@
+How to customize catalog promotion labels?
+==========================================
+
+To customize catalog promotion labels only what need to do is a overwrite proper template
```
Comment (GSadee) [inline docs/cookbook/promotions/customize-catalog-promotion-labels.rst:4]:
```suggestion
To customize catalog promotion labels only what need to do is overwrite the proper template
```

**Review by GSadee — APPROVED** [review]

File: docs/cookbook/promotions/catalog-promotion-labels.rst [hunk 13]
```diff
@@ -0,0 +1,113 @@
+How to customize catalog promotion labels?
+==========================================
+
+To customize catalog promotion labels only what need to do is overwrite the proper template
```
Comment (CoderMaggie) [inline docs/cookbook/promotions/catalog-promotion-labels.rst:4]:
```suggestion
To customize catalog promotion labels only what need to do is overwrite their template.
```

File: docs/cookbook/promotions/catalog-promotion-labels.rst [hunk 14]
```diff
@@ -0,0 +1,113 @@
...
+        {% for appliedPromotion in appliedPromotions %}
+            <div class="ui red big label promotion_label" style="margin: 0.5rem 0;">
+                <div class="row ui small sylius_catalog_promotion">
+                    {{ appliedPromotion.label }}{% if appliedPromotion.description and withDescription %} - {{ appliedPromotion.description }}{% endif %}
+                </div>
+            </div>
+        {% endfor %}
+    </div>
+
+After changes:
```
Comment (CoderMaggie) [inline docs/cookbook/promotions/catalog-promotion-labels.rst:29]:
```suggestion
How will it look after changes?
```

File: docs/cookbook/promotions/catalog-promotion-labels.rst [hunk 15]
```diff
@@ -0,0 +1,113 @@
...
+
+Product index:
+
+.. image:: ../../_images/cookbook/customization-catalog-promotion/catalog_promotion_label_product_index.png
+    :align: center
+
+Product details page:
```
Comment (CoderMaggie) [inline docs/cookbook/promotions/catalog-promotion-labels.rst:36]:
```suggestion
Product show:
```

**Review by CoderMaggie — APPROVED** [review]

**Review by GSadee — APPROVED** [review]

**Comment by GSadee** [comment]
The base of this pull-request was changed, you need fetch and reset your local branch 
if you want to add new commits to this pull request. **Reset before you pull, else commits
may become messed-up.**

Unless you added new commits (to this branch) locally that you did not push yet, 
execute `git fetch origin && git reset "catalog-promotion-label-customization"` to update your local branch.
 
Feel free to ask for assistance when you get stuck :+1:

**Comment by GSadee** [comment]
Thanks, Adam! :tada:

**Pull request merged**